Cross-node mail send blocked: runtime can't fetch related AFT record contract for inbox UPDATE

Cross-node QA 2026-05-02 reveals: even after delegate lazy-init (#78) and AFT contract delta-as-record fallback (#71 partial), bob never receives mail because alice's send is blocked at the **post-token-confirmation step** before the inbox UPDATE.

## Trace

\`\`\`
08:57:53.041  gw: Contract state updated  (alice's AFT record 7ytMm1EBbu... — token registered)
08:57:53.077  peer: Contract state updated  (bob's peer also got the state via UPDATE_PROPAGATION)
08:57:53.079  peer: erroneous verification with key prefix: e0b4be... @ update state (delta-as-record)
              ↑ sigverify fails on peer because resync flow uses a path that mishandles params
... no inbox UPDATE for GbxQZMzbJD6MzA9cSRAuuuYdHVRfdeXVEaxJCkMQMfwR after this
\`\`\`

Bob's inbox at \`GbxQZ...\` only ever saw the contract-init UPDATE_PROPAGATION at 08:47:15. After alice clicked Send + approved AFT permission, no further UPDATE for bob's inbox is propagated.

## Root cause hypothesis

\`MessageModel::start_sending\` queues the message in \`PENDING_INBOXES_UPDATE\`, then issues an AFT \`assign_token\` request. Inbox UPDATE only fires from \`AftRecords::confirm_allocation\` after the contract publishes a state summary that contains the new assignment.

In the live-node flow, alice's UI sees the gw publish the AFT update locally (state updated message), but \`confirm_allocation\` is gated on the UI's own subscription callback firing. The browser console shows:

\`\`\`
[ERROR] message not handled: ContractResponse(SubscribeResponse { ... subscribed: true })
\`\`\`

— suggesting the UI's WS handler doesn't dispatch \`SubscribeResponse\` events into \`confirm_allocation\`.

## Why the offline / single-node test passes

The Playwright \`example-data,no-sync\` build doesn't exercise the AFT round trip at all — messages are delivered via the in-memory mock. Only the live-node spec hits this path, and it's gated behind \`FREENET_LIVE_E2E_SEND=1\` because it never actually worked end-to-end.

## Repro

1. \`cargo make iso-nodes-up\` + republish webapp
2. Create alice on 7510, bob on 7511, share+import bob's contact in alice's address book
3. Alice sends mail to bob, approves AFT permission
4. Bob's inbox stays empty (\`0 messages\`) indefinitely

## Next steps

- Inspect \`ui/src/api.rs\` WS message handler to confirm it routes \`SubscribeResponse\` into \`AftRecords::confirm_allocation\` via the appropriate callback (or onto the \`StateSummary\` path)
- Fix routing so the inbox UPDATE actually fires after token mint
- Coupled with #71 / #72: even once inbox UPDATE fires, the cross-node sigverify / slot-collision issues are still open

--- a/ui/src/aft.rs
+++ b/ui/src/aft.rs
@@ -119,14 +119,6 @@ impl AftRecords {
         })
     }
 
-    /// Returns the locally-cached AFT record state for the given identity, if any.
-    /// Used by `MessageModel::finish_sending` to bundle the sender's AFT
-    /// record state alongside the inbox UPDATE delta as `RelatedStateAndDelta`,
-    /// bypassing the runtime's broken RequestRelated orchestration (#80).
-    pub fn record_state_for(identity: &Identity) -> Option<TokenAllocationRecord> {
-        RECORDS.with(|recs| recs.borrow().get(identity).cloned())
-    }
-
     pub async fn confirm_allocation(
         client: &mut WebApiRequestClient,
         aft_record: AftRecordId,
@@ -151,11 +143,13 @@ impl AftRecords {
             return Ok(());
         };
         // Optimistically append the freshly-minted assignment to the local
-        // RECORDS cache so `MessageModel::finish_sending` can bundle the
-        // post-burn state inline with the inbox UPDATE as
-        // `RelatedStateAndDelta` (see freenet/mail#80). Without this the
-        // local cache still reflects the pre-burn state and the receiving
-        // inbox contract sees an empty AFT record, rejecting the message.
+        // RECORDS cache so subsequent local reads (e.g. UI counters of
+        // remaining tokens, dev-mode introspection) reflect the post-burn
+        // state without having to wait for the AFT contract update to
+        // round-trip back through SubscribeResponse. Cross-node delivery
+        // does not depend on this cache anymore — runtime PR #4007 lets
+        // the inbox contract's `update_state` requires(AFT) signal resolve
+        // via `fetch_related_via_network` on the receiver side.
         if let Some(sender) = sender {
             RECORDS.with(|recs| {
                 let mut recs = recs.borrow_mut();

--- a/ui/src/inbox.rs
+++ b/ui/src/inbox.rs
@@ -301,42 +301,21 @@ impl MessageModel {
         });
 
         if let Some(update) = pending_update {
-            let token_record_id = assignment.token_record;
             let stored = update.msg.to_stored(assignment, &update.sender)?;
             let delta = UpdateInbox::AddMessages {
                 messages: vec![stored],
             };
             let delta_bytes = serde_json::to_vec(&delta)?;
-            // Bundle the sender's AFT record state alongside the inbox delta
-            // as `RelatedStateAndDelta` so the inbox contract can verify the
-            // burn locally without depending on the runtime's RequestRelated
-            // → GET orchestration (broken end-to-end, see #80).
-            let data = match AftRecords::record_state_for(&update.sender) {
-                Some(record) => {
-                    let related_state = record
-                        .serialized()
-                        .map_err(|e| format!("failed to serialize AFT record state: {e}"))?;
-                    UpdateData::RelatedStateAndDelta {
-                        related_to: token_record_id,
-                        state: related_state.into(),
-                        delta: delta_bytes.into(),
-                    }
-                }
-                None => {
-                    crate::log::error(
-                        format!(
-                            "no local AFT record state for sender `{}`, sending bare delta — \
-                             inbox contract will request it via RequestRelated",
-                            update.sender.alias()
-                        ),
-                        None,
-                    );
-                    UpdateData::Delta(delta_bytes.into())
-                }
-            };
+            // Send a bare `UpdateData::Delta` and let the runtime resolve
+            // the inbox contract's `update_state` requires(AFT record)
+            // signal via `fetch_related_via_network`. This relies on
+            // freenet-core PR #4006 (validate-side) + PR #4007
+            // (update-side) — without those the receiver's
+            // `update_state` errors with MissingRelated and the broadcast
+            // bounces through ResyncRequest recovery, see #80.
             let request = ContractRequest::Update {
                 key: inbox_contract,
-                data,
+                data: UpdateData::Delta(delta_bytes.into()),
             };
             client.send(request.into()).await?;
             // todo: event after sending, we may fail to update, must keep this in mind in case we receive no confirmation

